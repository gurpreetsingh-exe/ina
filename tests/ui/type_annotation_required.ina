// COMPILE_FAIL

type Type[T] = {}

impl[T] Type[T] {
    fn id[U](a: U) -> Type[U] {
        Type![U] {}
    }
}

fn test[T]() {}

fn main() {
    let a = Type {};
    test();
    // TODO: the error refers to the parameter `T` which belongs
    // to type `Type[T]` not the function `id`, so implement the
    // correct error at some point
    let a = Type::id(20);
}
