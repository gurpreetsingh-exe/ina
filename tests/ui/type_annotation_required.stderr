error: type annotation required
    ╭─[tests/ui/type_annotation_required.ina:19:13]
    │
 19 │     let a = Type::id(20);
    │             ────┬───
    │                 ╰───── cannot infer type of type parameter `T` in function `id`
    ∙
────╯
error: type annotation required
    ╭─[tests/ui/type_annotation_required.ina:14:13]
    │
 14 │     let a = Type {};
    │             ──┬─
    │               ╰─── cannot infer type of type parameter `T` in type `Type`
    ∙
────╯
error: type annotation required
    ╭─[tests/ui/type_annotation_required.ina:15:5]
    │
 15 │     test();
    │     ──┬─
    │       ╰─── cannot infer type of type parameter `T` in function `test`
    ∙
────╯
